renderer/src/common/contexts/prompt/provider.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState, type ReactNode } from 'react'
+import { PromptContext, type ReactHookFormPromptConfig } from '.'
+import { ReactHookFormPromptDialog } from './form-prompt-dialog'
+
+export function PromptProvider({ children }: { children: ReactNode }) {
```

Comment:
I am missing the term prompt 🤔 
Anyway why a context/provider approach is needed?

Reply:
no no, what I mean is, this implements something that is equivalent in usage/business logic to the native prompt, but implements it using shadcnui to match the design.

Simple pseudo code to compare

*Native*

```tsx
const groupName = prompt("Please enter group name", initialValue)

if (groupName) {
   createGroupMutation.mutate(groupName)
}
```

*Our implementation*

```tsx
const prompt = usePrompt()
const groupName = await prompt("Please enter group name", initialValue)

if (groupName) {
   createGroupMutation.mutate(groupName)
}
```

Reply:
I am honestly a bit confused, it seems unnecessary complication cause it is a simple dialog with a form that includes an input field. The things that confuse me are, the needs of use a context/provider for showing a simple dialog at root level and the term prompt 🤔 that is misleading comparing the component that is used under the hood